sinks/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,141 @@
+}
+
+type cloudwatchMetricSink struct {
+	logger              *logrus.Entry
+	client              *cloudwatch.Client
+	endpoint            string
+	region              string
+	namespace           string
+	standardUnitTagName types.StandardUnit
+}
+
+func NewCloudwatchMetricSink(endpoint, namespace, region string, standardUnitTagName types.StandardUnit, logger *logrus.Entry) *cloudwatchMetricSink {
+	return &cloudwatchMetricSink{
+		endpoint:            endpoint,
+		namespace:           namespace,
+		region:              region,
+		standardUnitTagName: standardUnitTagName,
+		logger:              logger,
+	}
+}
+
+func ParseConfig(name string, config interface{}) (veneur.MetricSinkConfig, error) {
+	cloudwatchConfig := CloudwatchMetricSinkConfig{}
+	err := util.DecodeConfig(name, config, &cloudwatchConfig)
+	if err != nil {
+		return nil, err
+	}
+	if cloudwatchConfig.CloudwatchStandardUnitTagName == "" {
+		cloudwatchConfig.CloudwatchStandardUnitTagName = DefaultCloudwatchStandardUnitTagName
+	}
+	return cloudwatchConfig, nil
+}
+
+func Create(
+	server *veneur.Server, name string, logger *logrus.Entry,
+	config veneur.Config, sinkConfig veneur.MetricSinkConfig,
+) (sinks.MetricSink, error) {
+	conf, ok := sinkConfig.(CloudwatchMetricSinkConfig)
+	if !ok {
+		return nil, errors.New("invalid sink config type")
+	}
+
+	return NewCloudwatchMetricSink(
+		conf.CloudwatchEndpoint,
+		conf.CloudwatchNamespace,
+		conf.AWSRegion,
+		conf.CloudwatchStandardUnitTagName,
+		logger,
+	), nil
+}
+
+func (s *cloudwatchMetricSink) Name() string {
+	return "cloudwatch"
+}
+
+func (s *cloudwatchMetricSink) Start(*trace.Client) error {
+	opts := cloudwatch.Options{}
+	if s.endpoint != "" {
+		opts.EndpointResolver = cloudwatch.EndpointResolverFromURL(s.endpoint)
+	}
+	if s.region != "" {
+		opts.Region = s.region
+	}
+	s.client = cloudwatch.New(opts)
+	return nil
+}
+
+func (s *cloudwatchMetricSink) Flush(ctx context.Context, metrics []samplers.InterMetric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+
+	metricData := make([]types.MetricDatum, len(metrics))
+	for i, metric := range metrics {
+		dimensions := make([]types.Dimension, 0, len(metric.Tags))
+		standardUnit := types.StandardUnitNone
+		for _, tag := range metric.Tags {
+			kv := strings.SplitN(tag, ":", 2)
+			if len(kv) < 2 {
+				continue // drop illegal tag
+			}
+			if kv[0] == string(s.standardUnitTagName) {
+				standardUnit = types.StandardUnit(kv[1])
+				continue
+			}
+			dimensions = append(dimensions, types.Dimension{
+				Name:  aws.String(kv[0]),
+				Value: aws.String(kv[1]),
+			})
+		}
+		metricData[i] = types.MetricDatum{
+			MetricName: aws.String(metric.Name),
+			Unit:       standardUnit,
+			Value:      aws.Float64(metric.Value),
+			Dimensions: dimensions,
+		}
+	}
+	input := &cloudwatch.PutMetricDataInput{
+		Namespace:  aws.String(s.namespace),
+		MetricData: metricData,
+	}
+	_, err := s.client.PutMetricData(ctx, input)
```

Comment:
I believe this context does not have a deadline associated with it. We should add a timeout that is configurable, but also maxes at the flush interval, similar to how the cortex sink works.

Reply:
This context has a deadline attached to it, from server.go `Flush`, but == that we may want to support a custom timeout <= the Veneur global flush interval.